kong/tracing/propagation.lua

```diff
@@ -397,6 +411,63 @@ local function parse_aws_headers(aws_header)
   return trace_id, span_id, should_sample
 end
 
+local function parse_gcp_headers(gcp_header)
+  local warn = kong.log.warn
+
+  if type(gcp_header) ~= "string" then
+    return nil, nil, nil
+  end
+
+  local trace_id, span_id, should_sample = match(gcp_header, GCP_TRACECONTEXT_PATTERN)
+
+  -- GCP header pattern is lenient on field existence and length
+  -- coalesce empty values to nil for easier validation
+  if trace_id == "" then
+    trace_id = nil
+  end
+  if span_id == "" then
+    span_id = nil
+  end
+  if should_sample == "" then
+    should_sample = nil
+  end
+
+  local invalid_values = {}
+  if trace_id == nil or #trace_id ~= GCP_TRACE_ID_LEN then
+    table.insert(invalid_values, "trace ID")
+  else
+    trace_id = from_hex(trace_id)
+  end
+
+  if span_id == nil or #span_id > GCP_SPAN_ID_MAX_LEN then
+    table.insert(invalid_values, "span ID")
+  else
+    -- the span id is a number between 1 and 20 digits long 
+    -- the propagation code expects span ids to be stored as hex strings.
+    -- left pad a 0 if the id has an odd number of digits for a valid hex value
+    if #span_id % 2 ~= 0 then
+      span_id = '0' .. span_id
+    end
+
+    span_id = from_hex(span_id)
```

Comment:
```suggestion

    span_id = from_dec(span_id):to_binary()
```
the span id is actually stored as binary data (from_hex converts a hex string to binary), in this case the span_id is represented as an unsigned integer so it should be returned as the output of: `from_dec(span_id):to_binary()`.
This may suggest a missing test case that ensures the span_id is treated correctly.